src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            var generator = editor.Generator;
+
+            var returns = node.DescendantNodesAndSelf().Where(n => n is TReturnStatementSyntax);
+            foreach (TReturnStatementSyntax returnSyntax in returns)
+            {
+                // Make sure we're not changing returns in nested local functions or lambdas
+                var containingMethod = returnSyntax.FirstAncestorOrSelf<SyntaxNode>(IsAsyncSupportingFunctionSyntax);
+                if (containingMethod != node)
+                {
+                    continue;
+                }
+
+                var returnExpression = generator.SyntaxFacts.GetExpressionOfReturnStatement(returnSyntax);
+                if (returnExpression is null)
+                {
+                    // Convert return; into return Task.CompletedTask;
+                    var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+                    editor.ReplaceNode(returnSyntax, returnTaskCompletedTask);
+                }
+                else
+                {
+                    // Convert return <expr>; into return Task.FromResult(<expr>);
+                    WrapExpressionWithTaskFromResult(returnExpression, editor, knownTypes);
+                }
+            }
+        }
+
+        private static void AppendTaskCompletedTaskReturn(SyntaxNode lastNode, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+            var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+
+            editor.InsertAfter(lastNode, returnTaskCompletedTask);
+        }
+
+        private static SyntaxNode GetReturnTaskCompletedTaskStatement(KnownTypes knownTypes, SyntaxGenerator generator)
+        {
+            var taskCompletedTaskInvocation = GetTaskCompletedTaskInvocation(knownTypes, generator);
+            var statement = generator.ReturnStatement(taskCompletedTaskInvocation);
+            return statement;
+        }
+
+        private static TExpressionSyntax GetTaskCompletedTaskInvocation(KnownTypes knownTypes, SyntaxGenerator generator)
+        {
+            var taskTypeExpression = TypeExpressionForStaticMemberAccess(generator, knownTypes._taskType);
+            return (TExpressionSyntax)generator.MemberAccessExpression(taskTypeExpression, nameof(Task.CompletedTask));
+        }
+
+        private static void WrapExpressionWithTaskFromResult(SyntaxNode expression, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+
+            var taskTypeExpression = TypeExpressionForStaticMemberAccess(generator, knownTypes._taskType);
+            var taskFromResult = generator.MemberAccessExpression(taskTypeExpression, nameof(Task.FromResult));
+            var invocation = generator.InvocationExpression(taskFromResult, expression.NormalizeWhitespace());
```

Comment:
def do not normalize the whitespace here.   taht will destroy personal user formatting.  What were you trying to use this for here?

Reply:
This causes a change in VB from:
```vb
    Dim f as Func(of Task) = Function() 1
```
to become:
```vb
    Dim f as Func(of Task) = Function() Task.FromResult(1
)
```

I can try just removing the trailing trivia from `expression`?

Reply:
ah.  the preferred things here is to take the original expression, and do .WithoutTrivia on it.  then take the new expression and do .WithTriviaFrom on it.  Effectively moving the original trivia to the outer call.

Reply:
Just moving Trivia causes loss of some trivia if there is non whitespace trivial on both tokens. It seems like there should be a helper method to move trivia between tokens/nodes when one is removed. This applies any time a token/node with significant trivia is removed.

Reply:
Example of issue
```VB
Dim zReminderRectangle As New Rectangle( ' Comment
  CInt( ' Comment 2
  (Width / 2) - (zTextSize.Height / 2)), CInt((Height / 2) - (zTextSize.Width / 2)), CInt(zTextSize.Height), CInt(zTextSize.Width))
```
becomes
```VB
Dim zReminderRectangle As New Rectangle( ' Comment
  (Width / 2) - (zTextSize.Height / 2), CInt((Height / 2) - (zTextSize.Width / 2)), CInt(zTextSize.Height), CInt(zTextSize.Width))
```
' Comment 2 is deleted
```C#
return /*Comment*/ (int) /* Comment 1*/ unchecked(currentKey * ((int)0xA5555529 + newKey));
```
becomes
```C#
return /*Comment*/  /* Comment 1*/ unchecked(currentKey * ((int)0xA5555529 + newKey));
```
C# does not lose the comment but has extra whitespace trivia
Should I just open a separate issue to cover all the places a token or node is removed?

Reply:
> Should I just open a separate issue to cover all the places a token or node is removed?

I think separate bugs for each code fixer that doesn't do the right thing with trivia would be ideal, as they probably have individual solutions, and small tasks can be more easily picked up by the community.

Reply:
> Should I just open a separate issue to cover all the places a token or node is removed?

Reply:
@CyrusNajmabadi did you mean to type something after the quote? 😛

Reply:
weird.  no clue what happened there :)